kafka/protocol/struct.py

```diff
@@ -12,14 +12,17 @@ class Struct(AbstractType):
     SCHEMA = Schema()
 
     def __init__(self, *args, **kwargs):
+        self.fields = {}
```

Comment:
I don't think we need this -- the same data is in `__dict__` and could be accessed via `getattr`. Or you might add a helper, like `get_field(name)` that returns `__dict__[name]`